src/bq_data_collector.py

```diff
@@ -0,0 +1,244 @@
+        repo_names = self._get_repo_list(eco_systems)
+
+        # Don't change this
+        year_prefix = '20*'
+        day_list = [item[2:] for item in last_n_days]
+        import pprint
+        pprint.pprint(day_list)
+        pprint.pprint(year_prefix)
+        query_params = {'{year_prefix_wildcard}': year_prefix,
+                        '{year_suffix_month_day}': '(' + ', '.join(["'" + d + "'" for d in day_list]) + ')',
+                        '{repo_names}': '(' + ', '.join(["'" + r + "'" for r in repo_names]) + ')', }
+        self._query_params, self._last_n_days = query_params, last_n_days
+
+    def get_gh_event_estimate(self):
+        query = """
+        SELECT  type as EventType, count(*) as Freq
+                FROM `githubarchive.day.{year_prefix_wildcard}`
+                WHERE _TABLE_SUFFIX IN {year_suffix_month_day}
+                AND repo.name in {repo_names}
+                AND type in ('PullRequestEvent', 'IssuesEvent')
+                GROUP BY type
+        """
+        query = bq_client_helper.bq_add_query_params(query, self._query_params)
+        return self._bq_client.query_to_pandas(query)
+
+    def get_issues_as_data_frame(self) -> pd.DataFrame:
+        """
+        Retrieves GH Issues as pandas data frame
+        """
+        return self._get_gh_event_as_data_frame(
+            {**self._query_params, **{'{payload_field_name}': 'issue', '{event_type}': 'IssuesEvent'}})
+
+    def get_prs_as_data_frame(self) -> pd.DataFrame:
+        """
+        Retrieves GH PRs as pandas data frame
+        """
+        return self._get_gh_event_as_data_frame(
+            {**self._query_params, **{'{payload_field_name}': 'pull_request', '{event_type}': 'PullRequestEvent'}})
+
+    def get_github_data(self) -> pd.DataFrame:
+        """
+        Retrives GH Issues and PRs and merge it into single dataframe
+        """
+        issues_df = self.get_issues_as_data_frame()
+        prs_df = self.get_prs_as_data_frame()
+
+        _logger.info('Merging issues and pull requests datasets')
+        cols = issues_df.columns
+        data_frame = pd.concat([issues_df, prs_df], axis=0, sort=False, ignore_index=True).reset_index(drop=True)
+        data_frame = data_frame[cols]
+
+        # update eco-system
+        if not data_frame.empty:
+            _logger.info('Updating eco-system')
+            data_frame['ecosystem'] = data_frame.apply(lambda x: self._update_eco_system(x['repo_name']),
+                                                       axis=1)
+
+        return data_frame
+
+    def _update_eco_system(self, repo_name):
+        """
+        Update ecosystem based on repo_name
+        """
+        if repo_name in self._golang_repo:
+            return 'golang'
+        elif repo_name in self._knative_repo:
+            return 'knative'
+        elif repo_name in self._kubevirt_repo:
+            return 'kubevirt'
+        else:
+            return None
+
```

Comment:
Few dependencies(e.g. golang/go) are common in all 3 ecosystems, in that case your latest implementation will end up tagging it wrong. I guess you may need to change the implementation or fallback to previous one.

Reply:
@arajkumar : in this case we can add ecosystem as  comm separated values Ex "golang,knative" etc. What do you think?

Reply:
This might work when the no.of ecosystems are low, but when it explodes it will be come cumbersome., but as of now you can go ahead with CSV for ecosystem field.

Reply:
As discussed added support for multiple eco-system. It will stored into 'ecosystem' column with comma separated.  Ex : "golang,knative"